jitbuilder/optimizer/JBOptimizer.cpp

```diff
@@ -117,11 +117,12 @@ static const OptimizationStrategy JBwarmStrategyOpts[] =
    { OMR::treeSimplification                                                       },
    { OMR::trivialDeadTreeRemoval,                    OMR::IfEnabled                },
 
-   { OMR::inductionVariableAnalysis,                 OMR::IfLoops                  },
-   { OMR::loopCanonicalization,                      OMR::IfLoops                  },
+   { OMR::basicBlockOrdering                                                       }, // clean up block order going into canonicalizer
+   { OMR::loopCanonicalization,                      OMR::IfLoops                  }, // canonicalizer first, otherwise indvar data gets messed up
```

Comment:
@0xdaryl I modified the text to hopefully make it more clear. I think each line's comment pertains to that optimization and shouldn't run on multiple lines, but I'm sure we're far from consistent on this point :( .